### PR TITLE
Refactor/itf fake peer prettify

### PR DESCRIPTION
**Pull request by MBoldyrev**
### Description of the Change

Unified the interface & renamed the test example.

### Benefits

Easier to use.

### Possible Drawbacks 

Bugs.

### CODE REVIEW

File: test/framework/integration_framework/fake_peer/fake_peer.hpp [hunk 1]
```diff
@@ -66,7 +69,8 @@ namespace integration_framework {
         shared_model::interface::AbstractTransportFactory<
             shared_model::interface::Transaction,
             iroha::protocol::Transaction>;
-    using YacStatePtr =
+    using MstMessagePtr = std::shared_ptr<MstMessage>;
```
Comment (l4l) [inline test/framework/integration_framework/fake_peer/fake_peer.hpp]: this type I guess can be shared via defining at the MstMessage file

File: test/integration/acceptance/CMakeLists.txt [hunk 2]
```diff
@@ -176,9 +176,9 @@ target_link_libraries(replay_test
     acceptance_fixture
     )
 
-addtest(basic_mst_state_propagation_test
-    basic_mst_state_propagation.cpp
+addtest(fake_peer_usage_example
```
Comment (l4l) [inline test/integration/acceptance/CMakeLists.txt]: All our tests has `_test` suffix, that's much easier to search them

File: test/framework/integration_framework/fake_peer/fake_peer.hpp (hunk 1 above)
Comment (MBoldyrev) [reply test/framework/integration_framework/fake_peer/fake_peer.hpp]: I would postpone this fix as in the next PR I move it from here anyway

File: test/framework/integration_framework/fake_peer/network/mst_network_notifier.hpp [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+  class MstNetworkNotifier final
+      : public iroha::network::MstTransportNotification {
+   public:
+    using MstMessagePtr = std::shared_ptr<MstMessage>;
+
+    void onNewState(const shared_model::crypto::PublicKey &from,
+                    const iroha::MstState &new_state) override;
+
+    rxcpp::observable<MstMessagePtr> get_observable();
```
Comment (lebdron) [inline test/framework/integration_framework/fake_peer/network/mst_network_notifier.hpp]: -> `getObservable`
please check other methods in this pull request

File: test/framework/integration_framework/fake_peer/fake_peer.cpp [hunk 4]
```diff
@@ -25,6 +28,9 @@
 using namespace shared_model::crypto;
 using namespace framework::expected;
 
+using YacStateMessage =
```
Comment (lebdron) [inline test/framework/integration_framework/fake_peer/fake_peer.cpp]: Looks redundant, `FakePeer::YacMessagePtr` can be used instead, please remove.

File: test/integration/acceptance/fake_peer_example_test.cpp [hunk 5]
```diff
@@ -82,26 +83,22 @@ class BasicMstPropagationFixture : public AcceptanceFixture {
...
-  auto notifications_getter =
-      std::make_shared<iroha::MockMstTransportNotification>();
   std::mutex mst_mutex;
   std::condition_variable mst_cv;
   std::atomic_bool got_state_notification(false);
-  EXPECT_CALL(*notifications_getter, onNewState(_, _))
-      .WillOnce(Invoke([&mst_cv, &got_state_notification](
-                           const auto &from_key, auto const &target_state) {
+  auto &itf = prepareState(1);
+  fake_peers_.front()->get_mst_states_observable().subscribe(
```
Comment (lebdron) [inline test/integration/acceptance/fake_peer_example_test.cpp]: 👍

File: test/integration/acceptance/fake_peer_example_test.cpp [hunk 6]
```diff
@@ -82,26 +83,22 @@ class BasicMstPropagationFixture : public AcceptanceFixture {
...
  * @then that peer propagates MST state to another peer
  */
-TEST_F(BasicMstPropagationFixture,
+TEST_F(FakePeerExampleFixture,
        MstStateOfTransactionWithoutAllSignaturesPropagtesToOtherPeer) {
-  auto notifications_getter =
-      std::make_shared<iroha::MockMstTransportNotification>();
   std::mutex mst_mutex;
   std::condition_variable mst_cv;
   std::atomic_bool got_state_notification(false);
```
Comment (lebdron) [inline test/integration/acceptance/fake_peer_example_test.cpp]: I hope this manual synchronization will be abstracted out in the end.

**Pull request merged**